Orange/widgets/evaluate/owpredictions.py

```diff
@@ -285,12 +300,16 @@ def _invalidate_predictions(self):
             self.predictors[inputid] = pred._replace(results=None)
 
     def _valid_predictors(self):
-        return [p for p in self.predictors.values()
-                if p.results is not None and not isinstance(p.results, str)]
+        if self.class_var is not None and \
+                self.data is not None:
+            return [p for p in self.predictors.values()
+                    if p.results is not None and not isinstance(p.results, str)]
+        else:
+            return []
 
     def _update_predictions_model(self):
         """Update the prediction view model."""
-        if self.data is not None:
+        if self.data is not None and self.class_var is not None:
```

Comment:
Doesn't the second part (class_var not None) also imply the first (data not None).

(The same `if` appears in a couple of places)